File: rust/2019/1-1.rs
use std::io::Read;

fn calculate_fuel(mass: i32) -> i32 {
    ((mass as f32 / 3.0).floor() - 2.0) as i32
}

fn solve(input: &String) -> i32 {
    input
        .lines()
        .map(|line| {
            let parsed = line.parse().expect("Failed to parse line");
            calculate_fuel(parsed)
        })
        .sum()
}

fn main() {
    let mut input = String::new();

    std::io::stdin()
        .read_to_string(&mut input)
        .expect("Failed to read stdin");

    println!("{}", solve(&input))
}

File: haskell/2019/1-1.hs
{-# LANGUAGE TypeApplications #-}

calculateFuel :: Int -> Int
calculateFuel mass = floor @Double (realToFrac mass / 3) - 2

solve :: String -> Int
solve = sum . fmap (calculateFuel . read) . lines

main :: IO ()
main = print . solve =<< getContents
